### ISSUE TITLE
Type declarations missing std namespace prefix

### ISSUE THREAD

**Issue by TheDoubleB** [ceres-solver/ceres-solver#209]
`HashMap` and `HashSet` type declarations are missing `std::` namespace prefixes for `map` and `set` base classes at the following file/lines:
https://github.com/ceres-solver/ceres-solver/blob/080ca4c5f2ac42620971a07f06d2d13deb7befa8/internal/ceres/collections_port.h#L82
https://github.com/ceres-solver/ceres-solver/blob/080ca4c5f2ac42620971a07f06d2d13deb7befa8/internal/ceres/collections_port.h#L85

**Comment by sandwichmaker** [comment]
what configuration - platform, compiler are you using that is triggering this error for you?

**Comment by TheDoubleB** [comment]
I compiled ceres-solver along with OpenMVG on Microsoft Windows using Visual Studio 2013 for x64 arch. I do not know why this configuration was triggered because it was a one-time thing, as it chose a different one when I cleared cmake cache.

But, in any case, when this configuration does get triggered, the "unknown base class" error for `HashMap` and `HashSet` will always be present because of the missing `std::` namespace prefix.
